checker/src/main/java/org/checkerframework/checker/index/inequality/LessThanTransfer.java

```diff
@@ -57,12 +60,26 @@ protected void refineGT(
                 // right is already bottom, nothing to refine.
                 return;
             }
-            lessThanExpressions.add(leftRec.toString());
+            if (isParsable(leftRec)) {
+                lessThanExpressions.add(leftRec.toString());
+            }
             Receiver rightRec = FlowExpressions.internalReprOf(analysis.getTypeFactory(), right);
             store.insertValue(rightRec, factory.createLessThanQualifier(lessThanExpressions));
         }
     }
 
+    /**
+     * Return true if {@code receiver}'s toString can be parsed by {@link FlowExpressionParseUtil}.
```

Comment:
I think it would be helpful for the comment to indicate that this is not a test of whether the expression is valid or well-formed (which was my first guess), but merely whether it is of a permitted type.  Also, returning true is not a promise that the expression is parseable -- but returning false means that it is not.  (At least, that's what I think.  I might be wrong, which is a reason to improve the documentation.)

Also:  It seems that this routine belongs in `FlowExpressionParseUtil` rather than in this class.